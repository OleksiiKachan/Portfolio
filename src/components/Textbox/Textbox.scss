@import '../../assets/style/variables';

.textbox {
  position: relative;
  &__input {
    width: 100%;
    padding: 10px 0;
    font-size: 0.9rem;
    color: $color-white;
    margin-bottom: 30px;
    border: none;
    border-bottom: 1px solid $color-white;
    outline: none;
    background-color: transparent;
    letter-spacing: 1px;

    &:focus ~ .textbox__label,
    &:valid ~ .textbox__label {
      top: -18px;
      left: 0;
      color: #03a9f4;
      font-size: 10px;
    }
  }

  &__label {
    position: absolute;
    top: 0;
    left: 0;
    padding: 10px 0;
    font-size: 0.8rem;
    color: $color-white;
    pointer-events: none;
    transition: all 0.5s ease-in-out;
    letter-spacing: 1px;
  }
}

@media screen and (max-width: $breakpoint-tablet) {
  .textbox {
    &__label {
      font-size: 0.7rem;
    }
  }
}
