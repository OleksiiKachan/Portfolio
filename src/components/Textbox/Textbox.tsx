import React from 'react';
import classNames from 'classnames';
import './Textbox.scss';

/*
    <Textbox
      fieldName=''
      label=''
      autoComplete
      required 
      className=''
    />
*/

type PropsType = {
  fieldName: string;
  label: string;
  type?: string;
  autoComplete?: boolean;
  required?: boolean;
  className?: string;
};

export default ({
  type = 'text',
  fieldName,
  label,
  autoComplete = false,
  required = false,
  className,
}: PropsType) => {
  return (
    <div className={classNames('textbox', className)}>
      <input
        id={fieldName}
        type={type}
        name={fieldName}
        autoComplete={autoComplete ? 'yes' : 'no'}
        required={required}
        className={classNames('textbox__input')}
      />
      <label className={classNames('textbox__label')} htmlFor={fieldName}>
        {label}
      </label>
    </div>
  );
};
