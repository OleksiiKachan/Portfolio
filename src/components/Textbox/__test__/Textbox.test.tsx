import React from 'react';
import { shallow } from 'enzyme';
import Textbox from '..';

describe('Textbox', () => {
  it('Snapshot', () => {
    const component = shallow(
      <Textbox fieldName="fieldName" label="Field Label" required />
    );

    expect(component).toMatchSnapshot();
  });
});
