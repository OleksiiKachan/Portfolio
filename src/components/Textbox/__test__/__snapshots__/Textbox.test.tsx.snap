// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textbox Snapshot 1`] = `
<div
  className="textbox"
>
  <input
    autoComplete="no"
    className="textbox__input"
    id="fieldName"
    name="fieldName"
    required={true}
    type="text"
  />
  <label
    className="textbox__label"
    htmlFor="fieldName"
  >
    Field Label
  </label>
</div>
`;
